### PR TITLE
Enforce order of imports.

### PR DESCRIPTION
**Pull request by matt-paul**
This PR is to create some logical order with our imports - the rules can be found in .eslintrc.json and seem reasonably sane from a react point of view - lifted from what i found other people out there are using.

NOTE: Server.js has a weird order of imports (from eslint perspective), and there doesn't seem massive value in adding extra rules just for that file, so i have ignored it.

What do people think?

### CODE REVIEW

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -21,9 +21,28 @@
...
+      {
+        "groups": ["builtin", "external", "internal"],
+        "pathGroups": [
+          {
+            "pattern": "react",
+            "group": "external",
+            "position": "before"
+          }
+        ],
+        "pathGroupsExcludedImportTypes": ["react"],
```
Comment (frigus02) [inline frontend/.eslintrc.json:38]: What does `pathGroups` and `pathGroupsExcludedImportTypes` do?

**Review by frigus02 — APPROVED** [review]
Looks good to me

File: frontend/.eslintrc.json (hunk 1 above)
Comment (matt-paul) [reply frontend/.eslintrc.json:38]: Puts react at the top

Comment (matt-paul) [reply frontend/.eslintrc.json:38]: Just because everyone is so used to that convention

Comment (frigus02) [reply frontend/.eslintrc.json:38]: Ah right. In my mind we don't need that. But it's also fine for me 🙂

**Comment by alsuren** [comment]
If VSCode's format-on-save is capable of applying this then go for it. On our previous project, we would have the problem that  vscode would auto-import something at the bottom of the imports list and then the linter would forbid committing because they weren't in the right order. This was infuriating.

The other thing that was infuriating was that vscode would auto-import a thing when you tab-complete it, and then not auto-remove it when you removed the code. This would also cause a linter error. Maybe I am conflating these two things in my head.

I suppose I mostly want to run eslint --fix on save or something. If that is possible then I'm happy to have as complicated linting rules as you want. Otherwise it will just make me grumpy. That said, I'm not the only person on the team, so if other people are more grumpy about unsorted imports then go for it.

**Comment by matt-paul** [comment]
> If VSCode's format-on-save is capable of applying this then go for it. On our previous project, we would have the problem that vscode would auto-import something at the bottom of the imports list and then the linter would forbid committing because they weren't in the right order. This was infuriating.
> 
> The other thing that was infuriating was that vscode would auto-import a thing when you tab-complete it, and then not auto-remove it when you removed the code. This would also cause a linter error. Maybe I am conflating these two things in my head.
> 
> I suppose I mostly want to run eslint --fix on save or something. If that is possible then I'm happy to have as complicated linting rules as you want. Otherwise it will just make me grumpy. That said, I'm not the only person on the team, so if other people are more grumpy about unsorted imports then go for it.


It nicely orders things on save for >95% of our uses.  The only one it struggled with was server.js.

**Comment by matt-paul** [comment]
```The other thing that was infuriating was that vscode would auto-import a thing when you tab-complete it, and then not auto-remove it when you removed the code. This would also cause a linter error. Maybe I am conflating these two things in my head.```

This is a separate issue.  I have autoimports turned off because it annoyed me and sometimes imported the wrong thing

**Pull request merged**